File: data/code_pairs/fortran/590.f90
module mod_operations
  implicit none
  type :: t
    integer :: x
  end type t
contains
  subroutine increment(x)
    type(t), intent(inout) :: x
    integer :: i
    do i = 1, 1000
      x%x = x%x + 1
    end do
  end subroutine increment
end module mod_operations

program main
  use mod_operations
  type(t) :: x
  x%x = 0
  call increment(x)
  if (x%x == 1000) then
    print *, 'Test passed.'
  else
    print *, 'Test failed.'
  endif
end program main

File: data/code_pairs/cpp/590.cpp
#include <iostream>
#include <stdexcept>

struct t {
    int x;
};

class Incrementer {
public:
    static void increment(t& x) {
        for (int i = 1; i <= 1000; ++i) {
            x.x += 1;
        }
    }
};

int main() {
    t x{0}; // Initialize x.x to 0
    Incrementer::increment(x);
    if (x.x == 1000) {
        std::cout << "Test passed: Value is " << x.x << std::endl;
    } else {
        std::cerr << "Test failed: Expected 1000, got " << x.x << std::endl;
        return 1;
    }
    return 0;
}